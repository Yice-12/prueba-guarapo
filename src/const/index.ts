export const SPECIE_LIST = [
  "Human",
  "Alien",
  "Humanoid",
  "Poopybutthole",
  "Mythological",
  "Unknown",
  "Animal",
  "Disease",
  "Robot",
  "Cronenberg",
];

export const GENDER_LIST = ["female", "male", "genderless", "unknown"];
export const STATUS_LIST = ["Alive", "Dead", "Unknown"];
