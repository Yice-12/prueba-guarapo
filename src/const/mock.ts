import { configureStore } from "@reduxjs/toolkit";
import { charactersSlice } from "../store/slices/charactersSlice/charactersSlice";

const mockState = {
  characters: {
    characters: [
      {
        id: 1,
        name: "Rick Sanchez",
        status: "Alive",
        species: "Human",
        type: "",
        gender: "Male",
        origin: {
          name: "Earth (C-137)",
          url: "https://rickandmortyapi.com/api/location/1",
        },
        location: {
          name: "Citadel of Ricks",
          url: "https://rickandmortyapi.com/api/location/3",
        },
        image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
        episode: ["https://rickandmortyapi.com/api/episode/1"],
        url: "https://rickandmortyapi.com/api/character/1",
        created: "2017-11-04T18:48:46.250Z",
      },
    ],
    loading: false,
    character: {
      id: 1,
      name: "Rick Sanchez",
      status: "Alive",
      species: "Human",
      type: "",
      gender: "Male",
      origin: {
        name: "Earth (C-137)",
        url: "https://rickandmortyapi.com/api/location/1",
      },
      location: {
        name: "Citadel of Ricks",
        url: "https://rickandmortyapi.com/api/location/3",
      },
      image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      episode: ["https://rickandmortyapi.com/api/episode/1"],
      url: "https://rickandmortyapi.com/api/character/1",
      created: "2017-11-04T18:48:46.250Z",
    },
    episodes: [
      {
        id: 1,
        name: "Pilot",
        air_date: "December 2, 2013",
        episode: "S01E01",
        characters: ["https://rickandmortyapi.com/api/character/1"],
        url: "https://rickandmortyapi.com/api/episode/1",
        created: "2017-11-10T12:56:33.798Z",
      },
    ],
    name: "",
  },
};

export const store = configureStore({
  reducer: {
    characters: charactersSlice.reducer,
  },
  preloadedState: mockState,
});
