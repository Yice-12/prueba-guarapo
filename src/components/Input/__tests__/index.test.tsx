import { fireEvent, render } from "@testing-library/react";
import { Input } from "../";
import { Provider } from "react-redux";
import { store } from "../../../const/mock";

test("render Input and change", () => {
  const mockSetPage = jest.fn();
  const { asFragment, getByPlaceholderText } = render(
    <Provider store={store}>
      <Input setPage={mockSetPage} />
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();

  const input = getByPlaceholderText("Filter by name...");

  fireEvent.change(input, { target: { value: "test" } });
});
