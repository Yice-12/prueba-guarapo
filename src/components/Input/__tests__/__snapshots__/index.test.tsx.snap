// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Input 1`] = `
<DocumentFragment>
  <div
    class="input"
  >
    <div
      class="input__content"
    >
      <img
        alt="loupe"
        src="[object Object]"
      />
      <input
        placeholder="Filter by name..."
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`render Input and change 1`] = `
<DocumentFragment>
  <div
    class="input"
  >
    <div
      class="input__content"
    >
      <img
        alt="loupe"
        src="[object Object]"
      />
      <input
        placeholder="Filter by name..."
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
